sh/abandon_chain.sh

```diff
@@ -0,0 +1,249 @@
+declare -a current_owners=("${owners_array[@]}")
+
+while (( ${#current_owners[@]} > 1 )) ; do
+    declare owner_to_remove="${current_owners[0]}"
+
+    declare removeOwner_call
+    removeOwner_call="$(cast calldata "$removeOwner_sig" "$sentinel" "$owner_to_remove" 1)"
+
+    calls+=(
+        "$(
+            cast concat-hex                                              \
+            0x00                                                         \
+            "$safe_address"                                              \
+            "$(cast to-uint256 0)"                                       \
+            "$(cast to-uint256 "$(((${#removeOwner_call} - 2) / 2))")"   \
+            "$removeOwner_call"
+        )"
+    )
+
+    # Update working state
+    current_owners=("${current_owners[@]:1}")
+done
+
+# Swap the last remaining owner with initial_owner
+declare last_owner="${current_owners[0]}"
+
+declare swapOwner_call
+swapOwner_call="$(cast calldata "$swapOwner_sig" "$sentinel" "$last_owner" "$initial_owner")"
+
+calls+=(
+    "$(
+        cast concat-hex                                            \
+        0x00                                                       \
+        "$safe_address"                                            \
+        "$(cast to-uint256 0)"                                     \
+        "$(cast to-uint256 "$(((${#swapOwner_call} - 2) / 2))")"   \
+        "$swapOwner_call"
+    )"
+)
+
+# Wrap in multiSend call
+declare multisend_calldata
+multisend_calldata="$(cast calldata "$multisend_sig" "$(cast concat-hex "${calls[@]}")")"
+declare -r multisend_calldata
+
+# set minimum gas price (mostly for Arbitrum and BNB)
+declare -i min_gas_price
+min_gas_price="$(get_config minGasPriceGwei)"
+min_gas_price=$((min_gas_price * 1000000000))
+declare -r -i min_gas_price
+declare -i gas_price
+gas_price="$(cast gas-price --rpc-url "$rpc_url")"
+if (( gas_price < min_gas_price )) ; then
+    echo 'Setting gas price to minimum of '$((min_gas_price / 1000000000))' gwei' >&2
+    gas_price=$min_gas_price
+fi
+declare -r -i gas_price
+declare -i gas_estimate_multiplier
+gas_estimate_multiplier="$(get_config gasMultiplierPercent)"
+declare -r -i gas_estimate_multiplier
+
+declare packed_signatures
+packed_signatures="$(retrieve_signatures abandon_chain "$multisend_calldata" 1)"
+declare -r packed_signatures
+
+# configure gas limit
+declare -r -a args=(
+    "$safe_address" "$execTransaction_sig"
+    # to, value, data, operation, safeTxGas, baseGas, gasPrice, gasToken, refundReceiver, signatures
+    "$multicall_address" 0 "$multisend_calldata" 1 0 0 0 "$(cast address-zero)" "$(cast address-zero)" "$packed_signatures"
+)
+
+# set gas limit and add multiplier/headroom (again mostly for Arbitrum)
+declare -i gas_limit
+gas_limit="$(cast estimate --from "$signer" --rpc-url "$rpc_url" --gas-price $gas_price --chain $chainid "${args[@]}")"
+gas_limit=$((gas_limit * gas_estimate_multiplier / 100))
+declare -r -i gas_limit
```

Comment:
Can you refer to the logic in the other scripts for handling EIP7825 and also reify/DRY that logic into one of the `common_*.sh` libraries?